### PR TITLE
fix virtual host issue #4

### PR DESCRIPTION
**Pull request by dandraka**
Hi @LordMike, first of all THANK YOU for your tool, it's a lifesaver :-) As others have noted, there is a small issue when trying to use a vhost name. I've tested this change extensively both on the default (/) vhost and named ones. Kind regards, Jim Andrakakis

### CODE REVIEW

File: src/MBW.Tools.RabbitDump/Movers/Amqp/UriConnectionFactory.cs [hunk 1]
```diff
@@ -23,9 +23,11 @@ public UriConnectionFactory(Uri uri)
                 Port = uri.IsDefaultPort ? 5672 : uri.Port,
                 UserName = user,
                 Password = password,
-                VirtualHost = uri.AbsolutePath,
+                VirtualHost = uri.AbsolutePath == "/" ? "/" : uri.AbsolutePath.Remove(0, 1),
                 AutomaticRecoveryEnabled = true
             };
+
+            Console.WriteLine($"Virtual host = '{_connectionFactoryImplementation.VirtualHost}'");
```
Comment (dandraka) [inline src/MBW.Tools.RabbitDump/Movers/Amqp/UriConnectionFactory.cs]: actually you can delete this line; it was debug info I forgot, apologies

Comment (dandraka) [reply src/MBW.Tools.RabbitDump/Movers/Amqp/UriConnectionFactory.cs]: deleted in next commit

**Comment by LordMike** [comment]
The issue I have with this is that the vhost doesn't need to be prefixed with `/` - but it can. If I f.ex. have a vhost that is named `/queue`, then I would have to use `amqp://server//queue`. Which looks bad..

So I'm not sure if the URI-approach is workable :/

**Comment by vadimart92** [comment]
@LordMike but virtual host *should* be prefixed with "/" char to comply with specification https://www.rabbitmq.com/uri-spec.html
May be you should consider using `ConnectionFactory.Uri` property instead of parsing parameters manually?

**Pull request merged**